src/kaleidoscope/plugin/Escape-OneShot.cpp

```diff
@@ -19,26 +19,30 @@
 #include <Kaleidoscope-OneShot.h>
 #include <Kaleidoscope-Escape-OneShot.h>
 #include "kaleidoscope/keyswitch_state.h"
+#include "kaleidoscope/layers.h"
 
 namespace kaleidoscope {
 namespace plugin {
 
-bool EscapeOneShot::did_escape_;
-
-EventHandlerResult EscapeOneShot::onKeyswitchEvent(Key &mapped_key, KeyAddr key_addr, uint8_t keyState) {
-  if (mapped_key != Key_Escape || (keyState & INJECTED))
-    return EventHandlerResult::OK;
-
-  if (did_escape_)
-    mapped_key = Key_NoKey;
-  did_escape_ = !keyToggledOff(keyState);
-
-  if ((!::OneShot.isActive() || ::OneShot.isPressed()) && !::OneShot.isSticky()) {
-    return EventHandlerResult::OK;
+EventHandlerResult EscapeOneShot::onKeyswitchEvent(Key &mapped_key, KeyAddr key_addr, uint8_t key_state) {
+  // We only act on an escape key that has just been pressed, and not
+  // generated by some other plugin. Also, only if at least one
+  // OneShot key is active and/or sticky. Last, only if there are no
+  // OneShot keys currently being held.
+  if (mapped_key == Key_Escape &&
+      keyToggledOn(key_state) &&
+      !(key_state & INJECTED) &&
+      ::OneShot.isActive() &&
+      !::OneShot.isPressed()) {
```

Comment:
This new test condition is required because the `pressed` flag wasn't being cleared when `sticky` OneShot keys were released. With that fixed in OneShot, it needed to change here.